src/browser/components/MattermostView.jsx

```diff
@@ -132,13 +132,16 @@ export default class MattermostView extends React.Component {
         } else if (destURL.path.match(/^\/help\//)) {
           // continue to open special case internal urls in default browser
           shell.openExternal(e.url);
-        } else {
+        } else if (Utils.isTeamUrl(this.props.src, e.url, true) || Utils.isPluginUrl(this.props.src, e.url)) {
           // New window should disable nodeIntegration.
           window.open(e.url, remote.app.getName(), 'nodeIntegration=no, contextIsolation=yes, show=yes');
+        } else {
+          e.preventDefault();
+          shell.openExternal(e.url);
         }
       } else {
-        // if the link is external, use default browser.
-        shell.openExternal(e.url);
+        // if the link is external, use default os' application.
+        ipcRenderer.send('confirm-protocol', destURL.protocol, e.url);
```

Comment:
we might want to add http/s as default safe, would that make sense? that way everything would go through the same mechanisms

Reply:
I'd be fine with that

Reply:
updated to handle http/s by default

Reply:
Could we simplify this code just a bit further by removing [this check](https://github.com/mattermost/desktop/blob/dabc2f571d7d1ab69446c3615853411721b077bd/src/browser/components/MattermostView.jsx#L110-L113) and switching [the comparison in `Utils.isInternalURL`](https://github.com/mattermost/desktop/blob/dabc2f571d7d1ab69446c3615853411721b077bd/src/utils/util.js#L40) to use origins instead of just hosts? And I guess `` `${scheme}:` `` would have to be added to the protocol whitelist, too. What do you think?

Reply:
ticket created for the URL vs url: https://mattermost.atlassian.net/browse/MM-21835